### PR TITLE
[core] Move the QIR peephole patterns out of tablegen.

### LINKED ISSUE #476: Rewrite tablegen rewrite patterns as C++ code
As part of code cleanup, it was decided to just write the C++ classes instead of trying to make use of tablegen to generate rewrite patterns. Peephole.td and Canonical.td are sets of rewrite patterns that should be converted from tablegen to C++ code for this cleanup.

### PR DESCRIPTION
**Pull request by schweitzpgi**
Moves all the peephole patterns to explicit patterns and out of tablegen.

Fix #476.

### CODE REVIEW

File: lib/Optimizer/CodeGen/VerifyQIRProfile.cpp [hunk 1]
```diff
@@ -9,7 +9,7 @@
 #include "PassDetails.h"
 #include "cudaq/Optimizer/Builder/Intrinsics.h"
 #include "cudaq/Optimizer/CodeGen/Passes.h"
-#include "cudaq/Optimizer/CodeGen/Peephole.h"
+#include "cudaq/Optimizer/CodeGen/QIRFunctionNames.h"
```
Comment (khalatepradnya) [inline lib/Optimizer/CodeGen/VerifyQIRProfile.cpp:12]: Clarifying question - why this change?

**Review by khalatepradnya — APPROVED** [review]
👍🏽

File: lib/Optimizer/CodeGen/VerifyQIRProfile.cpp (hunk 1 above)
Comment (schweitzpgi) [reply lib/Optimizer/CodeGen/VerifyQIRProfile.cpp:12]: The verifying doesn't need the peephole patterns (it doesn't rewrite the code), but it was including the peephole header to get some function names. It should've included the names directly.
